crates/forge-runner/src/test_case_setup.rs

```diff
@@ -0,0 +1,138 @@
+
+    cached_state
+        .state
+        .dict_state_reader
+        .class_hash_to_class
+        .insert(strk_class_hash, contract_class_strk());
+
+    let recipient = generate_random_felt();
+    let recipient_balance_low_address = map_entry_address("ERC20_balances", &[recipient]);
+    let recipient_balance_high_address =
+        StorageKey(recipient_balance_low_address.try_into().unwrap())
+            .next_storage_key()
+            .unwrap();
+    let total_supply_low = 55_401_946_922_417_748_965_830_181_u128;
+
+    // Update STRK storage to mimic constructor behavior
+    let storage_entries_and_values_to_update = [
+        // name
+        (
+            (
+                strk_contract_address,
+                storage_key(variable_address("ERC20_name")).unwrap(),
+            ),
+            Felt::from_short_string("STRK").unwrap(),
+        ),
+        // symbol
+        (
+            (
+                strk_contract_address,
+                storage_key(variable_address("ERC20_symbol")).unwrap(),
+            ),
+            Felt::from_short_string("STRK").unwrap(),
+        ),
+        // decimals
+        (
+            (
+                strk_contract_address,
+                storage_key(variable_address("ERC20_decimals")).unwrap(),
+            ),
+            Felt::from(18),
+        ),
+        // total_supply low
+        (
+            (
+                strk_contract_address,
+                storage_key(variable_address("ERC20_total_supply")).unwrap(),
+            ),
+            Felt::from(total_supply_low),
+        ),
+        // total_supply high
+        (
+            (
+                strk_contract_address,
+                storage_key(map_entry_address("ERC20_total_supply", &[Felt::ONE])).unwrap(),
+            ),
+            Felt::ZERO,
+        ),
+        // recipient balance low
+        (
+            (
+                strk_contract_address,
+                storage_key(recipient_balance_low_address).unwrap(),
+            ),
+            Felt::from(total_supply_low),
+        ),
+        // recipient balance high
+        (
+            (
+                strk_contract_address,
+                storage_key(**recipient_balance_high_address).unwrap(),
+            ),
+            Felt::ZERO,
+        ),
+        // permitted_minter
+        (
+            (
+                strk_contract_address,
+                storage_key(variable_address("permitted_minter")).unwrap(),
+            ),
+            generate_random_felt(),
+        ),
+        // skip initializing roles
+        // upgrade_delay
+        (
+            (
+                strk_contract_address,
+                storage_key(variable_address("upgrade_delay")).unwrap(),
+            ),
+            Felt::ZERO,
+        ),
+        // TODO: Decide if we want to write `domain_hash` to storage
+        // it enforces us to read chain_id if the test uses forking, hence
+        // this is a potential performance issue
```

Comment:
note: Looking at the code related with domain hash, I don't think it's actually needed. As mentioned in the comment, it would enforce from us to fetch chain id in every test case which could have vast impact on making requests to provider.

Reply:
Can you expand on why do we need to fetch chain id?

Reply:
Sure, in [this line](https://github.com/starknet-io/starkgate-contracts/blob/2c2729d47b9b5c7421c16fd258b2fa7e3ad6ddfc/src/cairo/strk/erc20_lockable.cairo#L142), we write domains hash to storage, which uses [`calc_domains_hash`](https://github.com/starknet-io/starkgate-contracts/blob/2c2729d47b9b5c7421c16fd258b2fa7e3ad6ddfc/src/cairo/strk/eip712helper.cairo#L52) and in its implementation you can see it needs chain id.